mpf/devices/segment_display/segment_display.py

```diff
@@ -155,11 +165,56 @@ def add_text_entry(self, text, color, flashing, flash_mask, transition, transiti
-                                           color)
-        self._update_display(SegmentDisplayState(text, flashing, flash_mask))
+
+        ###############################
+        #Store current color and text as previous text/color of next run even if no transition in this step,
+        #the next step might have a transition, that the old text/color needs to be included into that transition
+        ###############################
+
+        # Handle new and previous text
+        if self._previous_text:
+            previous_text = self._previous_text
+        else:
+            previous_text = ""
+        self._previous_text = text  # Save the new text as the next previous text
+
+        # Handle new and previous color
+        if self._previous_color:
+            previous_color = self._previous_color
+        else:
+            previous_color = self._default_color
+        self._previous_color = color  # Save the new color as the next previous color
+
+        if transition or self._previous_transition_out:
+            if transition: #if transition exists, then ignore transition_out of previous text/color
+                transition_conf = TransitionManager.get_transition(self.size,
+                                                                   self.config['integrated_dots'],
+                                                                   self.config['integrated_commas'],
+                                                                   self.config['use_dots_for_commas'],
+                                                                   transition)
+            elif self._previous_transition_out:
+                transition_conf = TransitionManager.get_transition(self.size,
+                                                                   self.config['integrated_dots'],
+                                                                   self.config['integrated_commas'],
+                                                                   self.config['use_dots_for_commas'],
+                                                                   self._previous_transition_out)
+                self._previous_transition_out = None # Once the transistion_out is played removed it that is not played in the next step again
+            if transition_out:  #in case transition_out is set we need to preserve it for the next step but only after the previous transition_out is in this step's config
+                self._previous_transition_out = transition_out
+
+            #start transition
+            self._start_transition(transition_conf, previous_text, text,
+                                   previous_color, color,
+                                   self.config['default_transition_update_hz'], flashing, flash_mask)
+
+        else: #No transition configured
+            if transition_out:  #in case transition_out is set we need to preserve it for the next step
```

Comment:
The gotcha here is that if there is no `transition_out` on _this_ transition but the previous transition _did_ have an out, then `self._previous_transition_out` will still have the value of the last transition out. Best to be explicit and not use an if statement.

```
self._previous_transition_out = transition_out or None
```

Which is also in the above if block, so it could be called just once at the end of this method outside of the if/else block.